dask/base.py

```diff
@@ -60,3 +62,27 @@ def compute(*args, **kwargs):
     keys = [arg._keys() for arg in args]
     results = get(dsk, keys, **kwargs)
     return tuple(a._finalize(a, r) for a, r in zip(args, results))
+
+
+normalize = Dispatch()
+normalize.register(object, lambda a: a)
+normalize.register(dict, lambda a: tuple(sorted(a.items())))
+with ignoring(ImportError):
+    import pandas as pd
+    normalize.register(pd.DataFrame, lambda a: (id(a), len(a), list(a.columns)))
+    normalize.register(pd.Series, lambda a: (id(a), len(a), a.name))
+with ignoring(ImportError):
+    import numpy as np
+    normalize.register(np.ndarray, lambda a: (id(a), a.dtype, a.shape))
```

Comment:
A theme of mine is that I'm generally against making technology or abstractions until absolutely necessary.  I may take this too far though.